tonic/models/vic/grid_params.py

```diff
@@ -1036,64 +1441,153 @@ def veg(veg_file, soil_dict, max_roots=3, veg_classes=11,
+                albedo[cell, vind, :] = np.array(line).astype(float)
+                row += 1
         cell += 1
     veg_dict = OrderedDict()
     veg_dict['gridcell'] = gridcel[:cell]
     veg_dict['Nveg'] = nveg[:cell]
     veg_dict['Cv'] = cv[:cell, :]
-    veg_dict['root_depth'] = root_depth[:cell, :]
-    veg_dict['root_fract'] = root_fract[:cell, :]
+    veg_dict['root_depth'] = root_depth[:cell, :, :]
+    veg_dict['root_fract'] = root_fract[:cell, :, :]
 
     if blowing_snow:
         veg_dict['sigma_slope'] = sigma_slope[:cell, :]
         veg_dict['lag_one'] = lag_one[:cell, :]
         veg_dict['fetch'] = fetch[:cell, :]
 
-    if lai_index:
+    if vegparam_lai and lai_src == FROM_VEGPARAM:
         veg_dict['LAI'] = lai[:cell, :, :]
 
+    if vegparam_fcan and fcan_src == FROM_VEGPARAM:
+        veg_dict['fcanopy'] = fcan[:cell, :, :]
+
+    if vegparam_albedo and alb_src == FROM_VEGPARAM:
+        veg_dict['albedo'] = albedo[:cell, :, :]
+
+    # Make gridcell order match that of soil_dict
     inds = []
     for sn in soil_dict['gridcell']:
         inds.append(np.nonzero(veg_dict['gridcell'] == sn))
 
+    new_veg_dict = OrderedDict()
     for var in veg_dict:
-        veg_dict[var] = np.squeeze(veg_dict[var][inds])
-    return veg_dict
+        new_veg_dict[var] = np.squeeze([veg_dict[var][i] for i in inds])
+
+    return new_veg_dict
+# -------------------------------------------------------------------- #
 
 
 # -------------------------------------------------------------------- #
-def veg_class(veg_file, maxcols=57, skiprows=3, c=Cols()):
+def lake(lake_file, soil_dict, max_numnod=10,
+        cells=False, lake_profile=False):
     """
-    Load the entire vegetation library file into a dictionary of numpy arrays.
-    Also reorders data to match gridcell order of soil file.
+    Read the lake file from lakeFile.  Assumes max length for depth-area
+    relationship.  Also reorders data to match gridcell order of soil file.
     """
 
-    print('reading {0}'.format(veg_file))
+    print('reading {0}'.format(lake_file))
+
+    with open(lake_file) as f:
+        lines = f.readlines()
 
-    usecols = np.arange(maxcols)
+    if not cells:
+        cells = len(lines)
 
-    data = np.loadtxt(veg_file, usecols=usecols, skiprows=skiprows)
+    gridcel = np.zeros(cells).astype(int)
+    lake_idx = np.zeros(cells).astype(int)
+    numnod = np.zeros(cells).astype(int)
+    mindepth = np.zeros(cells)
+    wfrac = np.zeros(cells)
+    depth_in = np.zeros(cells)
+    rpercent = np.zeros(cells)
+    if lake_profile:
+        basin_depth = np.zeros((cells, max_numnod))
+        basin_area = np.zeros((cells, max_numnod))
+    else:
+        basin_depth = np.zeros(cells)
+        basin_area = np.zeros(cells)
 
-    veglib_dict = OrderedDict()
-    for var in c.veglib:
-        veglib_dict[var] = np.squeeze(data[:, c.veglib[var]])
+    row = 0
+    cell = 0
+    while row < len(lines):
```

Comment:
I think you can use a for loop here and cut down on some of the indexing logic below: 

`for cell, line in enumerate(lines):`